src/components/Asset/AssetActions/Trade/Swap.tsx

```diff
@@ -56,148 +61,182 @@ export default function Swap({
+          ...prevState,
+          amount: amountDataToken.toString(),
+          maxAmount: maximumDt.toString()
+        }))
+      } catch (error) {
+        LoggerInstance.error(error.message)
+      }
     }
     calculateMaximum()
   }, [
-    asset,
-    maxOcean,
     maxDt,
+    maxBaseToken,
     balance,
-    values?.type,
+    values.type,
     setMaximumDt,
-    setMaximumOcean
+    setMaximumBaseToken,
+    assetExtended,
+    web3,
+    dtItem.token,
+    baseTokenItem.token
   ])
 
   const switchTokens = () => {
     setFieldValue('type', values.type === 'buy' ? 'sell' : 'buy')
-    setCoin(values.type === 'sell' ? 'OCEAN' : asset.datatokens[0].symbol)
+    setCoin(
+      values.type === 'sell' ? 'OCEAN' : assetExtended.datatokens[0].symbol
+    )
     // don't reset form because we don't want to reset type
     setFieldValue('datatoken', 0)
-    setFieldValue('ocean', 0)
+    setFieldValue('baseToken', 0)
     setErrors({})
   }
 
   const handleValueChange = async (name: string, value: number) => {
-    const tokenIn = ''
-    const tokenOut = ''
+    let tokenIn = ''
+    let tokenOut = ''
+    const poolInstance = new Pool(web3)
     let newValue
 
-    // if (name === 'ocean') {
-    //   if (values.type === 'sell') {
-    //     newValue = await ocean.pool.getDTNeeded(price.address, value.toString())
-
-    //     setTotalValue(newValue)
-    //     setTokenAmount(value.toString())
-
-    //     tokenIn = ddo.services[0].datatokenAddress
-    //     tokenOut = ocean.pool.oceanAddress
-    //   } else {
-    //     newValue = await ocean.pool.getDTReceived(
-    //       price.address,
-    //       value.toString()
-    //     )
-
-    //     setTotalValue(value.toString())
-    //     setTokenAmount(newValue)
-    //     tokenIn = ocean.pool.oceanAddress
-    //     tokenOut = ddo.services[0].datatokenAddress
-    //   }
-    // } else {
-    //   if (values.type === 'sell') {
-    //     newValue = await ocean.pool.getOceanReceived(
-    //       price.address,
-    //       value.toString()
-    //     )
-
-    //     setTotalValue(value.toString())
-    //     setTokenAmount(newValue)
-    //     tokenIn = ddo.services[0].datatokenAddress
-    //     tokenOut = ocean.pool.oceanAddress
-    //   } else {
-    //     newValue = await ocean.pool.getOceanNeeded(
-    //       price.address,
-    //       value.toString()
-    //     )
-
-    //     setTotalValue(newValue)
-    //     setTokenAmount(value.toString())
-    //     tokenIn = ocean.pool.oceanAddress
-    //     tokenOut = ddo.services[0].datatokenAddress
-    //   }
-    // }
-
-    await setFieldValue(name === 'ocean' ? 'datatoken' : 'ocean', newValue)
-
-    // const spotPrice = await ocean.pool.getSpotPrice(
-    //   price.address,
-    //   tokenIn,
-    //   tokenOut
-    // )
-
-    // setSpotPrice(spotPrice)
+    if (name === 'baseToken') {
+      if (values.type === 'sell') {
+        newValue = await poolInstance.calcPoolOutGivenSingleIn(
```

Comment:
the `calc....` functions are for liquidity add/remove calculation. We need to use `getAmountInExactOut ` and `getAmountOutExactIn` here